share/pacer/Pacer.cpp

```diff
@@ -0,0 +1,201 @@
+        }
+        else {
+            IsInitialized = false;
+            std::cerr << "[ERROR] Pacer: GPTL is not initialized in PACER_INTEGRATED mode." << std::endl;
+            return false;
+        }
+    }
+
+    return true;
+}
+
+/// Start the time named TimerName
+bool Pacer::start(const std::string &TimerName)
+{
+    PACER_CHECK_INIT();
+
+    PACER_CHECK_ERROR(GPTLstart(TimerName.c_str()));
+
+    auto it = OpenTimers.find(TimerName);
+    if (it != OpenTimers.end() )
+        OpenTimers[TimerName]++;
+    else
+        OpenTimers[TimerName] = 1;
+    return true;
+}
+
+/// Stop the time named TimerName
+/// Issues warning if timer hasn't been started yet
+bool Pacer::stop(const std::string &TimerName)
+{
+    PACER_CHECK_INIT();
+
+    auto it = OpenTimers.find(TimerName);
+
+    if (it != OpenTimers.end() ) {
+        PACER_CHECK_ERROR(GPTLstop(TimerName.c_str()));
+
+        if ( OpenTimers[TimerName] == 1 )
+            OpenTimers.erase(TimerName);
+        else
+            OpenTimers[TimerName]--;
+    }
+    else {
+        std::cerr << "[WARNING] Pacer: Trying to stop timer: \""
+            << TimerName << "\" before starting it." << std::endl;
+
+        return false;
+    }
+
+    return true;
+}
+
+/// Sets named prefix for all subsequent timers
+bool Pacer::setPrefix(const std::string &Prefix)
+{
+    PACER_CHECK_INIT();
+
+    PACER_CHECK_ERROR(GPTLprefix_set(Prefix.c_str()));
+
+    return true;
+}
+
+/// Unsets prefix for all subsequent timers
+bool Pacer::unsetPrefix()
+{
+    PACER_CHECK_INIT();
+
+    PACER_CHECK_ERROR(GPTLprefix_unset());
+
+    return true;
+}
+
+/// Prints timing statistics and global summary files
+/// Output Files: TimerFilePrefix.timing.<MyRank>
+/// TimerFilePrefix.summary
+/// PrintAllRanks: flag to control if per rank timing files are printed
+bool Pacer::print(const std::string &TimerFilePrefix, bool PrintAllRanks /*= = false */)
+{
+    PACER_CHECK_INIT();
+
+    std::string TimerFileName = TimerFilePrefix + ".timing." + std::to_string(MyRank);
+    std::string SummaryFileName = TimerFilePrefix + ".summary";
+
+    PACER_CHECK_ERROR(GPTLpr_summary_file(InternalComm, SummaryFileName.c_str()));
+
+    if ( PrintAllRanks == false ) {
+        if (MyRank == 0) {
+            PACER_CHECK_ERROR(GPTLpr_file(TimerFileName.c_str()));
+        }
+    }
+    else
+        PACER_CHECK_ERROR(GPTLpr_file(TimerFileName.c_str()));
+
+    return true;
+}
+
+/// Cleans up Pacer
+/// Issues warning if any timers are still open
+bool Pacer::finalize()
+{
+    PACER_CHECK_INIT();
+
+    if ( PacerMode == PACER_STANDALONE )
+        PACER_CHECK_ERROR(GPTLfinalize());
+
+    if ( (MyRank == 0) && ( OpenTimers.size() > 0) ){
+        std::cerr << "[WARNING] Pacer: Following " << OpenTimers.size() << " timer(s) is/are still open." << std::endl;
+        for (auto i = OpenTimers.begin(); i != OpenTimers.end(); i++)
+            std::cerr << '\t' << i->first << std::endl;
+    }
+
```

Comment:
Might want a OpenTimers.clear() call just to make sure the static map is cleared of all entries on finalization.

Reply:
Done.